modules/remotecontrol/Controller.js

```
@@ -0,0 +1,136 @@
+        this.area.dblclick(
+            this._onMouseClickHandler.bind(this, "mousedblclick"));
+        this.area.contextmenu(() => false);
+        this.area[0].onmousewheel = event => {
+            this._sendEvent({
+                type: "mousescroll",
+                x: event.deltaX,
+                y: event.deltaY
+            });
+        };
+        $(window).keydown(this._onKeyPessHandler.bind(this, "keydown"));
+        $(window).keyup(this._onKeyPessHandler.bind(this, "keyup"));
+    }
+
+    /**
+     * Stops processing the mouse and keyboard events.
+     */
+    stop() {
+        this.area.off( "mousemove" );
+        this.area.off( "mousedown" );
+        this.area.off( "mouseup" );
+        this.area.off( "contextmenu" );
+        this.area.off( "dblclick" );
+        $(window).off( "keydown");
+        $(window).off( "keyup");
+        this.area[0].onmousewheel = undefined;
+    }
+
+    /**
+     * Handler for mouse click events.
+     * @param {String} type the type of event ("mousedown"/"mouseup")
+     * @param {Event} event the mouse event.
+     */
+    _onMouseClickHandler(type, event) {
+        this._sendEvent({
+            type: type,
+            button: event.which
+        });
+    }
+
+    /**
+     * Handler for key press events.
+     * @param {String} type the type of event ("keydown"/"keyup")
+     * @param {Event} event the key event.
+     */
+    _onKeyPessHandler(type, event) {
+        this._sendEvent({
+            type: type,
+            key: getKey(event),
+            modifiers: getModifiers(event),
+        });
+    }
+
+    /**
+     * Sends remote control event to the controlled participant.
+     * @param {Object} event the remote control event.
+     */
+    _sendRemoteControlEvent(event) {
+        try{
+            APP.conference.sendEndpointMessage("",
+                {type: "remote-control-event", event});
+        } catch (e) {
+            // failed to send the event.
```

Why errors are ignored ?